crates/solver/src/driver.rs

```diff
@@ -420,13 +446,54 @@ impl Driver {
                     .map_err(|err| anyhow!("{err}"))
                     .context("convert nonce")?,
             };
+            let scores = model::solver_competition::Scores {
+                winner: scores.peek().expect("no winner").0,
+                winning_score: scores.next().expect("no score").1, // guaranteed to exist
+                // reference score is the second highest score, or 0 if there is only one score (see
+                // CIP20)
+                reference_score: scores.next().unwrap_or_default().1,
+                block_deadline: {
+                    let deadline = self.solver_time_limit
+                        + self.solution_submitter.max_confirm_time
+                        + self.additional_mining_deadline;
+                    let number_blocks = deadline.as_secs() / self.block_time.as_secs();
+                    block_during_simulation + number_blocks
+                },
+            };
+            let participants = solver_competition
+                .solutions
+                .iter()
+                .map(|solution| solution.solver_address)
+                .collect::<HashSet<_>>(); // to avoid duplicates
+                                          // external prices for all tokens contained in the trades of a settlement
```

Comment:
`participants` is now a `HashMap` so we no longer have to be descriptive here.
```suggestion
                .collect();
            // external prices for all tokens contained in the trades of a settlement
```

Reply:
Removed.